xrft/tests/test_xrft.py

```
@@ -568,6 +568,12 @@ def test_cross_phase_true_phase_2d(self, dask):
         )  # Offset in [-pi, pi]
         npt.assert_almost_equal(offset, phase_offset)
 
+    def test_scaling_density_flag(self):
+        N = 25
+        s = xr.DataArray(np.random.rand(N), dims="x", coords={"x": range(N)})
+        with pytest.raises(ValueError):
+            ps = xrft.power_spectrum(s, dim="x", scaling="spectrum", density=True)
```

Great, thanks for adding this test. Could you also add a case for `density=False` and `scaling='density'`?

---

In this condition, only a warning will be raised. Should we test for a warning ?

---

Ah ok, no need for testing the warning then.

---

Ok, it think it should raise an error too. I will correct that

---

Can you add the case for `density=False` and `scaling='spectrum'`?

---

Actually, since the default is not going to be set as `scaling='density'`, I think the case of `densitt=False` and `scaling='density'` should only be a warning and not an error.

---

I like your original idea of when `density=False`, the `scaling` flag being ignored.

---

Ok, so let's go for that. I just add the other test for the raised error